core-bundle/src/Resources/contao/controllers/FrontendIndex.php

```diff
@@ -247,7 +247,7 @@ public function renderPage($pageModel)
 		}
 
 		// Check wether the language matches the root page language
-		if (isset($_GET['language']) && Config::get('addLanguageToUrl') && Input::get('language') != $objPage->rootLanguage)
+		if (isset($_GET['language']) && !$objPage->urlPrefix && Input::get('language') != $objPage->rootLanguage)
```

Comment:
Are you sure that this is correct? Because `addLanguageToUrl` is the equivalent of `$objPage->urlPrefix` and not `!$objPage->urlPrefix` IMHO.

Reply:
```suggestion
		if (isset($_GET['language']) && $objPage->urlPrefix !== '' && Input::get('language') != $objPage->rootLanguage)
```
This should be correct then?

Reply:
I guess but I haven't tested it. Not sure if the `!== ''` is a good idea, because the value might be `null`.

Reply:
```suggestion
		if (isset($_GET['language']) && $objPage->urlPrefix && Input::get('language') != $objPage->rootLanguage)
```

Reply:
Done in aee185e8a07e37885d3edc62deadb5b6a01caa1f